packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/template/ObjectFieldTemplate.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+  ADDITIONAL_PROPERTY_FLAG,
+  canExpand,
+  FormContextType,
+  getTemplate,
+  getUiOptions,
+  ObjectFieldTemplatePropertyType,
+  ObjectFieldTemplateProps,
+  RJSFSchema,
+  StrictRJSFSchema,
+} from '@rjsf/utils'
+import React, { useRef } from 'react'
+import { Button, Tooltip } from '@mui/material'
+import { useDeepCompareEffectNoCheck } from 'use-deep-compare-effect'
+import { displayToast } from '../../ToastMessage'
+import AddToList from '../../../assets/icons/AddToList'
+
+/**
+ * Derived from the base ObjectFieldTemplate with annotations-editor-specific changes
+ * - Custom button for adding additional properties
+ * - Tracks properties to show a toast message if a property is converted to an additionalProperty because it was
+ *   dropped from the schema (e.g. in a conditional schema)
+ *
+ * See https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/mui/src/ObjectFieldTemplate/ObjectFieldTemplate.tsx
+ * @param props
+ * @returns
+ */
+export function ObjectFieldTemplate<
+  T = any,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any,
+>(props: ObjectFieldTemplateProps<T, S, F>) {
+  const {
+    description,
+    disabled,
+    formData,
+    idSchema,
+    onAddClick,
+    properties,
+    readonly,
+    registry,
+    required,
+    schema,
+    title,
+    uiSchema,
+  } = props
+
+  const options = getUiOptions<T, S, F>(uiSchema)
+  const TitleFieldTemplate = getTemplate<'TitleFieldTemplate', T, S, F>(
+    'TitleFieldTemplate',
+    registry,
+    options,
+  )
+  const DescriptionFieldTemplate = getTemplate<
+    'DescriptionFieldTemplate',
+    T,
+    S,
+    F
+  >('DescriptionFieldTemplate', registry, options)
+
+  /*
+     Tracks the property keys that were last known to exist in the object.
+     */
+  const previousSchemaDefinedProperties = useRef<Set<string>>(new Set())
```

Comment:
Track this in a ref instead of useState to prevent multiple checks from occurring in strict mode, causing the toast to be displayed multiple times.